compiler/rustc_session/src/config.rs

```diff
@@ -1039,6 +1039,9 @@ fn default_configuration(sess: &Session) -> CrateConfig {
     if sess.opts.debug_assertions {
         ret.insert((sym::debug_assertions, None));
     }
+    if sess.overflow_checks() {
```

Comment:
I didn't see any FCP, so it will first need to be gated for nightly-only.
```suggestion
    if sess.is_nightly_build() && sess.overflow_checks() {
```

Reply:
Please do not make features default-on in nightly. Use feature gates instead.

Reply:
We already have some other feature gated cfgs, see `GATED_CFGS` for examples.